### PR TITLE
New settings page UI

### PR DESCRIPTION
**Pull request by Tabrisrp**

### CODE REVIEW

File: views/settings/fields/import-form.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	<div class="error"><p><?php _e( 'Before you can upload your import file, you will need to fix the following error:', 'rocket' ); ?></p>
+	<p><strong><?php echo esc_html( $data['upload_dir']['error'] ); ?></strong></p></div>
+<?php
+} else {
+	?>
+	<form action="<?php echo admin_url( 'admin-post.php' ); ?>" method="POST" enctype="multipart/form-data" class="wpr-tools">
+		<div class="wpr-tools-col">
+			<label for="upload" class="wpr-title3 wpr-tools-label wpr-icon-import"><?php _e( 'Import settings', 'rocket' ); ?></label>
+			<div class="wpr-upload">
+				<input type="file" id="upload" name="import" size="25" />
```
Comment (Screenfeed) [inline views/settings/fields/import-form.php]: Could be nice: `accept=".txt,.json"`.

File: views/settings/fields/one-click-addon.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+				<?php esc_html_e( 'Add-on status', 'rocket' ); ?>
+			</label>
+		</div>
+	</div>
+</div>
+
+<div class="wpr-field wpr-addon">
+	<div class="wpr-flex">
+		<div class="wpr-addon-logo">
+			<!--<?php echo esc_attr( $data['logo'] ); ?>-->
```
Comment (Screenfeed) [inline views/settings/fields/one-click-addon.php]: Hmm?

File: views/settings/fields/rocket-addon.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+				<?php esc_html_e( 'Add-on status', 'rocket' ); ?>
+			</label>
+		</div>
+	</div>
+</div>
+
+<div class="wpr-field wpr-addon">
+	<div class="wpr-flex">
+		<div class="wpr-addon-logo">
+			<!--<?php echo esc_attr( $data['logo'] ); ?>-->
```
Comment (Screenfeed) [inline views/settings/fields/rocket-addon.php]: Hmm²?

File: views/settings/page-sections/cloudflare.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+		</h2>
+	</div>
+	<?php $this->render_settings_sections( $data['id'] ); ?>
+	<div class="wpr-optionHeader">
+		<h3 class="wpr-title2"><?php esc_html_e( 'Cloudflare Cache', 'rocket' ); ?></h3>
+	</div><br>
+	<p class="wpr-field-description">
+	<?php
+	// translators: %s is the URL to the CloudFlare documentation.
+	printf( __( 'Purges cached resources for your website. <a href="%s" target="_blank">Learn more</a>', 'rocket' ), 'https://support.cloudflare.com/hc/en-us/articles/200169246' );
```
Comment (Screenfeed) [inline views/settings/page-sections/cloudflare.php]: I'd prefer see something like this:
```php
printf(
	// translators: %s is a "Learn more" link.
	__( 'Purges cached resources for your website. %s', 'rocket' ),
	'<a href="' . esc_url( __( 'https://support.cloudflare.com/hc/en-us/articles/200169246', 'rocket' ) ) . '" target="_blank">' . __( 'Learn more', 'rocket' ) . '</a>'
);
```

File: views/settings/page-sections/dashboard.php [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+	<div class="wpr-Page-row">
+		<div class="wpr-Page-col">
+			<div class="wpr-optionHeader">
+				<h3 class="wpr-title2"><?php esc_html_e( 'Frequently Asked Questions', 'rocket' ); ?></h3>
+			</div>
+			<fieldset class="wpr-fieldsContainer-fieldset">
+				<div class="wpr-field">
+					<ul class="wpr-field-list">
+					<?php foreach ( $data['faq'] as $faq_item ) : ?>
+						<li class="wpr-icon-information"><a href="<?php echo esc_attr( $faq_item['url'] ); ?>" data-beacon-article="<?php echo esc_attr( $faq_item['id'] ); ?>"><?php echo esc_html( $faq_item['title'] ); ?></a></li>
```
Comment (Screenfeed) [inline views/settings/page-sections/dashboard.php]: `esc_url()` for `$faq_item['url']`.

File: views/settings/page-sections/license.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+<div class="wpr-sectionHeader">
+	<h2 id="<?php echo esc_attr( $data['id'] ); ?>" class="wpr-title1 wpr-icon-important"><?php echo esc_html( $data['title'] ); ?></h2>
+	<div class="wpr-sectionHeader-title wpr-title3">
+		<?php _e( 'WP Rocket was not able to automatically validate your license.', 'rocket' ); ?>
+	</div>
+	<div class="wpr-sectionHeader-description">
+		<?php
+		// translators: %1$s = tutorial URL, %2$s = support URL.
+		printf( __( 'Follow this <a href="%1$s">tutorial</a>, or contact <a href="%2$s">support</a> to get the engine started.', 'rocket' ), '', '' );
```
Comment (Screenfeed) [inline views/settings/page-sections/license.php]: The URLs are missing.

File: views/settings/partials/sidebar.php [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+?>
+
+<h3 class="wpr-Sidebar-title wpr-title2"><?php _e( 'How to correctly measure your website’s loading time?', 'rocket' ); ?></h3>
+<div class="wpr-Sidebar-notice">
+	<p><?php _e( 'Learn how to use best practices to correctly measure your site\'s performance.', 'rocket' ); ?></p>
+	<a href="https://wp-rocket.me/blog/correctly-measure-websites-page-load-time/" target="_blank" class="wpr-Sidebar-notice-link"><?php _e( 'Read our guide', 'rocket' ); ?></a>
+</div>
+<div class="wpr-Sidebar-notice">
+	<p><?php _e( 'Why Google PageSpeed grade should not matter', 'rocket' ); ?></p>
+	<a href="https://wp-rocket.me/blog/the-truth-about-google-pagespeed-insights/" target="_blank" class="wpr-Sidebar-notice-link"><?php _e( 'Read more', 'rocket' ); ?></a>
```
Comment (Screenfeed) [inline views/settings/partials/sidebar.php]:
```
<a href="<?php echo esc_url( __( 'https://wp-rocket.me/blog/the-truth-about-google-pagespeed-insights/', 'rocket' ) ); ?>" target="_blank" class="wpr-Sidebar-notice-link">
```

File: views/settings/partials/sidebar.php [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+<?php if ( ! get_rocket_option( 'user_cache', 0 ) ) : ?>
+<div class="wpr-Sidebar-info">
+	<i class="wpr-icon-information2"></i>
+	<h4><?php _e( 'You have not activated logged-in user cache.', 'rocket' ); ?></h4>
+	<p><?php _e( 'Use a private browser to check your website\'s speed and visual appearance.', 'rocket' ); ?></p>
+</div>
+<?php endif; ?>
+<?php $this->render_part( 'documentation' ); ?>
+
+<!-- <button data-beacon-id="" class="wpr-button wpr-icon-help wpr-button--lightBlue wpr-button--fixed"><?php esc_html_e( 'Need Help?', 'rocket' ); ?></button> -->
```
Comment (Screenfeed) [inline views/settings/partials/sidebar.php]: Hmm³?

File: inc/classes/admin/settings/class-page.php [hunk 9]
```diff
@@ -167,7 +190,259 @@ public function render_page() {
...
+
+		if ( ! current_user_can( apply_filters( 'rocket_capability', 'manage_options' ) ) ) {
+			wp_die();
+		}
+
+		delete_transient( 'wp_rocket_customer_data' );
+
+		$customer_data = $this->get_customer_data();
+
+		set_transient( 'wp_rocket_customer_data', $customer_data, DAY_IN_SECONDS );
```
[the remark this answers is not in the document]
Comment (Tabrisrp) [reply inc/classes/admin/settings/class-page.php]: I don't understand what you mean here

File: inc/classes/admin/settings/class-render.php [hunk 10]
```diff
@@ -85,11 +77,33 @@ function( array $item ) {
...
+
+		$default = [
+			'id'               => '',
+			'title'            => '',
+			'menu_description' => '',
+			'class'            => '',
+		];
+
+		$navigation = array_map(
+			function( array $item ) use ( $default ) {
```
[the remark this answers is not in the document]
Comment (Tabrisrp) [reply inc/classes/admin/settings/class-render.php]: in fact I need it, else `$default` doesn't exist in the context of the closure

File: inc/classes/admin/settings/class-render.php [hunk 11]
```diff
@@ -312,4 +466,91 @@ public function render_import_form() {
...
+			'parameter'  => '',
+			'attributes' => '',
+		];
+
+		$args = wp_parse_args( $args, $default );
+
+		if ( ! empty( $args['attributes'] ) ) {
+			$attributes = '';
+			foreach ( $args['attributes'] as $key => $value ) {
+				$attributes .= ' ' . sanitize_key( $key ) . '="' . esc_attr( $value ) . '"';
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply inc/classes/admin/settings/class-render.php]: Isn't this the same?
```php
		$navigation = array_map(
			function( array $item ) {
				$item = wp_parse_args( $item, [
					'id'               => '',
					'title'            => '',
					'menu_description' => '',
					'class'            => '',
				] );

				if ( ! empty( $item['class'] ) ) {
					$item['class'] = implode( ' ', array_map( 'sanitize_html_class', $item['class'] ) );
				}

				unset( $item['sections'] );
				return $item;
			},
			$navigation
		);
```

File: inc/classes/admin/settings/class-render.php (hunk 10 above)
[the remark this answers is not in the document]
Comment (Screenfeed) [reply inc/classes/admin/settings/class-render.php]: Isn't this the same?
```php
		$navigation = array_map(
			function( array $item ) {
				$item = wp_parse_args( $item, [
					'id'               => '',
					'title'            => '',
					'menu_description' => '',
					'class'            => '',
				] );

				if ( ! empty( $item['class'] ) ) {
					$item['class'] = implode( ' ', array_map( 'sanitize_html_class', $item['class'] ) );
				}

				unset( $item['sections'] );
				return $item;
			},
			$navigation
		);
```

**Pull request merged**